packages/contracts-bedrock/test/invariants/echidna/superc20/PROPERTIES.md

```diff
@@ -0,0 +1,41 @@
+supERC20 properties
+===================
+
+Valid state
+-----------
+
+| id    | description                                                           | halmos | echidna |
+| ----- | -----                                                                 | -----  | -----   |
+|   0   | calls to sendERC20 succeed as long as caller has enough balance       |        |         |
+
+Variable transition
+-------------------
+
+| id    | description                                                           | halmos | echidna |
+| ----- | -----                                                                 | -----  | -----   |
+|   1   | sendERC20 decreases the token's totalSupply in the source chain       |        |         |
+|   2   | relayERC20 increases the token's totalSupply in the destination chain |        |         |
+|   3   | only calls to `convert()` can increase the total supply across chains |        |         |
+
+High level
+----------
+
+| id    | description                                                                                                  | halmos | echidna |
+| ----- | -----                                                                                                        | -----  | -----   |
+|   4   | sum of total supply across all chains is always `<=` to `convert()`ed amount                                 |        |         |
```

Comment:
```suggestion
|   4   | sum of total supply across all chains is always `<=` to `convert()`-ed and `mint()`-ed amount                                 |        |         |
```

Reply:
I'm not sure I follow you on this one, what `mint()` are you referring to?

with this property I'm trying to describe the high-level behaviour of the whole system, where the only place where tokens can be minted should be inside `convert`

I guess we could also define a separate 'Valid State' property local to the supERC20 where we check total supply is the sum of all mints, if we don't get that for free by including  ToB ERC20 properties

Reply:
I'm referring to the `OptimismSuperchainERC20#mint` function, that is only callable by the standard bridge on the L1 -> L2 flow.
Should we take that into consideration when defining the total amount of the token invariant?